cinema/urls.py

```diff
@@ -1,10 +1,43 @@
-from django.urls import path
+from rest_framework import routers
+from django.urls import path, include
 
-from cinema.views import movie_list, movie_detail
+from cinema.views import (
+    ActorList,
+    ActorDetail,
+    GenreList,
+    GenreDetail,
+    MovieViewSet,
+    CinemaHallViewSet,
+)
+
+router = routers.DefaultRouter()
+router.register("movies", MovieViewSet)
 
 urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path("genres/", GenreList.as_view(), name="genre-list"),
+    path("genres/<int:pk>/", GenreDetail.as_view(), name="genre-detail"),
+    path("actors/", ActorList.as_view(), name="actor-list"),
+    path("actors/<int:pk>/", ActorDetail.as_view(), name="actor-detail"),
+    path(
+        "cinema_halls/",
+        CinemaHallViewSet.as_view(
+            actions={"get": "list", "post": "create"}
+        ),
+        name="cinema-hall-list"
+    ),
+    path(
+        "cinema_halls/<int:pk>/",
+        CinemaHallViewSet.as_view(
+            actions={
+                "get": "retrieve",
+                "put": "update",
+                "patch": "partial_update",
+                "delete": "destroy",
+            }
+        ),
+        name="cinema-hall-detail"
+    ),
```

Comment:
Assigning views configurations to variables and then using these variables in the urlpatterns can give you a much cleaner overall look of the url patterns.